include/alpaka/core/UniqueId.hpp

```diff
@@ -73,13 +73,13 @@ namespace alpaka
                 static constexpr std::size_t value = N;
             };
 
-#ifdef BOOST_COMP_MSVC
+#if (BOOST_COMP_MSVC || __CUDACC__)
```

Comment:
Yes the second pass is not compiling with nvcc. Currently I changed everything that it compiles on GPU and CPU without errors. 
I checked  both version with gcc and both are working fine. NVCC compiles but create no unique ids.